src/library/commissioner_impl.cpp

```diff
@@ -205,6 +209,11 @@ Error CommissionerImpl::ValidateConfig(const Config &aConfig)
     if (aConfig.mEnableCcm)
     {
         tlv::Tlv domainNameTlv{tlv::Type::kDomainName, aConfig.mDomainName};
+
+#if !OT_COMM_CONFIG_CCM_ENABLE
+        ExitNow(error = ERROR_INVALID_ARGS("CCM teatures not supported"));
```

Comment:
Thanks!

There is guideline that static objects with non-trivial destructor should be avoid. I defined an macro `CCM_NOT_IMPLEMENTED` to avoid repeating the error message:
```C++
#define CCM_NOT_IMPLEMENTED "CCM features not implemented"
```